### PR TITLE
[JENKINS-25315] Restore nested view owner fields

### PR DESCRIPTION
**Pull request by jglick**
[JENKINS-25315](https://issues.jenkins-ci.org/browse/JENKINS-25315)

### CODE REVIEW

**Comment by jglick** [comment]
@reviewbybees

**Comment by jglick** [comment]
@lvotypko you may be interested in reviewing this too. Are you maintaining this plugin now, or still @ctapobep?

**Comment by ctapobep** [comment]
No, I'm not maintaining it anymore.

**Comment by oleg-nenashev** [comment]
:bee: 
The test is hardly readable, but it's better than nothing

**Comment by amuniz** [comment]
:bee:

**Comment by jglick** [comment]
@reviewbybees done

**Comment by olivergondza** [comment]
Dirty, but I do not see a better way to go, too.

**Comment by rendner** [comment]
Are there any plans to release this fix in the near future?

**Comment by jglick** [comment]
Not sure if this plugin has an owner. I have been waiting a long time for movement on this & #24.

**Comment by lvotypko** [comment]
Hello, I am afraid that there is no maintainer. I can adopt it as a passive maintainer - check pull requests and do releases until someone is interested to adopt it.

**Comment by jglick** [comment]
Ah I see it is already up for adoption. Yes that would be helpful if you can spare a few minutes.

**Comment by lvotypko** [comment]
The changes looks fine - not a nice solution, but I do not see any better, too.

File: src/main/java/hudson/plugins/nested_view/NestedView.java [hunk 1]
```diff
@@ -92,6 +93,16 @@ public NestedView(String name) {
         super(name);
     }
 
+    private Object readResolve() throws Exception {
+        for (View view : views) {
+            // Unfortunately various methods which set View.owner are inappropriate here.
+            Field ownerF = View.class.getDeclaredField("owner");
```
Comment (rendner) [inline src/main/java/hudson/plugins/nested_view/NestedView.java]: Is there a reason for retrieving and manipulating the owner field (multiple times) inside the for loop? This should be done before the for loop.

File: src/main/java/hudson/plugins/nested_view/NestedView.java [hunk 2]
```diff
@@ -92,6 +93,16 @@ public NestedView(String name) {
         super(name);
     }
 
+    private Object readResolve() throws Exception {
```
Comment (rendner) [inline src/main/java/hudson/plugins/nested_view/NestedView.java]: Where is the new method readResolve called?

**Pull request merged**